tensorrt_llm/_torch/auto_deploy/transformations/library/sharding.py

```diff
@@ -57,6 +57,13 @@ class ShardingTransformInfo(BaseModel, ABC):
     rank: int
     world_size: int
 
+    def validate(self, gm: GraphModule = None, node: Node = None) -> bool:
```

Comment:
I think in the future it would be cleaner to separate the config class from the actual utility functions like this one. 

I prefer having a cleaner split especially if it becomes user-facing for a couple of reasons:

1. Serializability: simple config classes are usually serializable off-the-shelf, i.e., they only consist of basic data types (str, int, float, dict, list). This makes writing config parser super easy. 
2. Documentation. Simple config classes when they are user-facing are easier to read
3. Separation of concerns: having one class responsible for executing and one responsible for configuring ensures better design as it is more natural to keep a clean separation.

You don't need to change the design in this PR - we can merge it as is and consider this for the next set of PRs